docs/src/content/docs/reference/plugins.md

```diff
@@ -163,70 +245,75 @@ The example above will log a message that includes the provided info message:
 [long-process-plugin] Starting long process…
 ```
 
-#### `injectTranslations`
+#### `useTranslations`
 
-**type:** `(translations: Record<string, Record<string, string>>) => void`
+**type:** `(lang: string) => I18nT`
 
-A callback function to add or update translation strings used in Starlight’s [localization APIs](/guides/i18n/#using-ui-translations).
-
-In the following example, a plugin injects translations for a custom UI string named `myPlugin.doThing` for the `en` and `fr` locales:
+A callback function to generate a utility function to access UI strings for a given language.
+To learn more about the generated utility function and all the APIs available, see the [“Using UI translations”](/guides/i18n/#using-ui-translations) guide.
 
-```ts {6-13} /(injectTranslations)[^(]/
+```ts {6}
 // plugin.ts
 export default {
-  name: 'plugin-with-translations',
+  name: 'plugin-use-translations',
   hooks: {
-    setup({ injectTranslations }) {
-      injectTranslations({
-        en: {
-          'myPlugin.doThing': 'Do the thing',
-        },
-        fr: {
-          'myPlugin.doThing': 'Faire le truc',
-        },
-      });
+    setup({ useTranslations, logger }) {
+      const t = useTranslations('en');
+      logger.info(t('builtWithStarlight.label'));
     },
   },
 };
 ```
 
-To use the injected translations in your plugin UI, follow the [“Using UI translations” guide](/guides/i18n/#using-ui-translations).
+The example above will log a message that includes a built-in UI string for the English language:
 
-Types for a plugin’s injected translation strings are generated automatically in a user’s project, but are not yet available when working in your plugin’s codebase.
-To type the `locals.t` object in the context of your plugin, declare the following global namespaces in a TypeScript declaration file:
+```shell
+[plugin-use-translations] Built with Starlight
+```
 
-```ts
-// env.d.ts
-declare namespace App {
-  type StarlightLocals = import('@astrojs/starlight').StarlightLocals;
-  // Define the `locals.t` object in the context of a plugin.
-  interface Locals extends StarlightLocals {}
-}
+#### `pathToLang`
 
-declare namespace StarlightApp {
-  // Define the additional plugin translations in the `I18n` interface.
-  interface I18n {
-    'myPlugin.doThing': string;
-  }
-}
-```
+**type:** `(path: string) => string`
 
-You can also infer the types for the `StarlightApp.I18n` interface from a source file if you have an object containing your translations.
+A callback function to get the language for a given full file path.
```

Comment:
Aligning this with the previous suggestion, which I also like because it makes clear the arguments to pass as well as the purpose.

```suggestion
Call `pathToLang()` with a file path to get the language for that file.
```

One question here though: in general these docs always say “full” file path. What happens if I do `pathToLang('src/content/docs/en/foo.md')`? Or even `pathToLang('en/foo.md')`? Is it dependent on being an absolute path?

Reply:
I can add a few test cases for these, but they all fallback to the default locale.